linkcheck/httputil.py

```diff
@@ -44,6 +44,7 @@ def x509_to_dict(x509):
     }
     notAfter = x509.get_notAfter()
     if notAfter is not None:
+        notAfter = notAfter.decode()
```

Comment:
There's no regression test for this fix.  :(

Reply:
Indeed the limited https testing is completely disabled. There's a lot of code around certificates using sockets which probably needs ripping out and replacing with getting the certificate from requests. Best left until after the bs4 changes though.
This change is just a result of the pyOpenSSL API - which itself recommends using cryptography, and that would give us a nice datetime!

Reply:
Looks like there is no simple way to get the certificate from requests. Session.cert is the default client certificate (must update my local copy of the requests docs...).

Reply:
I've included this commit with a test in #338.